gutenTAG/base_oscillations/custom_input.py

```diff
@@ -0,0 +1,73 @@
+from .interface import BaseOscillationInterface
+from ..utils.global_variables import BASE_OSCILLATION_NAMES, TIMESTAMP
+from ..utils.types import BOGenerationContext
+from ..utils.default_values import default_values
+
+
+class CustomInput(BaseOscillationInterface):
+    KIND = BASE_OSCILLATION_NAMES.CUSTOM_INPUT
+
+    def get_base_oscillation_kind(self) -> str:
+        return self.KIND
+
+    def get_timeseries_periods(self) -> Optional[int]:
+        return None
+    
+    def generate_only_base(self,
+                        ctx: BOGenerationContext,
+                        length: Optional[int] = None,
+                        input_timeseries_path_train: Optional[str] = None,
+                        input_timeseries_path_test: Optional[str] = None,
+                        usecols: Optional[List[str]] = None,
+                        semi_supervised: Optional[bool] = None,
+                        supervised: Optional[bool] = None,
+                        *args, **kwargs) -> np.ndarray:
+        
+        """
+             Generate a numpy array using the data from the input timeseries file.
+
+             Args:
+                 ctx (BOGenerationContext): The context object for base object generation.
+                 length (Optional[int]): The desired length of the generated array. If not specified, default is self.length.
+                 input_timeseries_path_train (Optional[str]): The path to the input timeseries file for training data. If not specified, default is self.input_timeseries_path_train.
+                 input_timeseries_path_test (Optional[str]): The path to the input timeseries file for test data. If not specified, default is self.input_timeseries_path_test.
+                 usecols (Optional[List[str]]): The list of column names to be returned from the CSV file. If not specified, default is None, which returns all columns.
+                 semi_supervised (Optional[bool]): A flag indicating whether the generation is for semi-supervised learning. If not specified, default is None.
+                 supervised (Optional[bool]): A flag indicating whether the generation is for supervised learning. If not specified, default is None.
+
+             Returns:
+                 np.ndarray: A 1-dimensional or multi-dimensional numpy array, depending on the selected columns.
+
+             If `usecols` is not provided, or if it is `None`, the function will return all columns in the input CSV file. In this case, the dimensionality of the returned array will be the same as the original dataframe, which could be 1-dimensional or multi-dimensional, depending on the structure of the CSV file.
+
+             If `usecols` is provided with a list of column names that selects only one column, the returned array will be 1-dimensional.
+
+             If `usecols` is provided with a list of column names that selects multiple columns, the returned array will be multi-dimensional, with one dimension for each selected column.
```

Comment:
This is not true. According to L70: `df.iloc[:, 0]`, you return only the first column. This does make sense from a programming perspective, though:
All GutenTAG base oscillation so far generate a single channel (1-dimensional array) on each call. If you want to generate multiple channels, you have to call them multiple times. I would argue that this should also be the case for the `custom_input` base oscillation. In addition, I would also have two column-selection similar to the input paths:

Please refactor `usecols`:
- new name: `use_column_train` and `use_column_test`
- new type: `Union[str, int]` (this means, we can either select a column by its name or position)
- both `input_timeseries_path_test` and `use_column_test` are required

Reply:
Thanks for all the comments and explanations. It helped me understand and get to the source of the issues.
I hope that my latest changes bring the feature close to where we want it to be.